### PR TITLE
week 5 assessment answers

### PR DESCRIPTION
**Pull request by JoshuaMaurer**
# Submitting Week 5 Assessment

### ✅ Completed Files

Mark the following as confirmation that the files in this repository are complete.

- [ ✅] JavaScript Coding Challenges
- [ ✅] Ruby Coding Challenges
- [ ✅] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:
Thanks Gene and as always would love any and all constructive feedback and improvements!

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -20,33 +20,361 @@ const secretCodeWord2 = "Gobbledygook"
 // Expected output: "G0bbl3dyg00k"
 const secretCodeWord3 = "Eccentric"
 // Expected output: "3cc3ntr1c"
+const secretCodeWord4 = "EeEe"
+const secretCodeWord5 = "Supercalifragilisticexpialidocious"
+
+// input: a string
+
+// output: a string with the following input letters changed to numbers. "a" to 4, "e" to 3, "i" to 1, and "o" to 0
+
```
Comment (TsunamiCode5) [inline code-challenges.test.js:29]: Surprise! Not Gene but I hope you enjoy my feedback as well. AMAZING pseudocode here!

File: code-challenges.test.js [hunk 2]
```diff
@@ -20,33 +20,361 @@ const secretCodeWord2 = "Gobbledygook"
...
+// pseudo:
+// Create a function called superSecretCodingMachine that takes a string as a parameter. Use .replaceAll() to replace all instances of the searched letter with the regex search pattern. Use /g to search all occurences of the letter in the string and /i to be case insensitive, then use a comma to separate the search pattern from the replacement value and add corresponding replacement value. Repeat this .replaceALL pattern three additional times, changing to the required corresponding searched letter and its number replacement. Return in one line off the single passed through string parameter.
+
+// test:
+// describe("superSecretCodingMachine", () => {
+//     it("take in a string and returns a string with a coded message" , () => {
+//         const secretCodeWord1 = "Lackadaisical"
+//         const secretCodeWord2 = "Gobbledygook"
+//         const secretCodeWord3 = "Eccentric"
```
Comment (TsunamiCode5) [inline code-challenges.test.js:38]: GREAT hoisting! This makes tests look so much cleaner and it's efficient as well.

File: code-challenges.test.js [hunk 3]
```diff
@@ -20,33 +20,361 @@ const secretCodeWord2 = "Gobbledygook"
...
+        "E": 3,
+        "i": 1,
+        "o": 0,
+        "u": 6,
+      };
+    return string.replaceAll(/a|e|E|i|o|u/gi, (matched) => superSecretCode[matched])
+}
+
+// This one below was done AFTER initial assessment submission. It has a couple additional things cleaned up. The one above passes but I was not happy with it. 
+
```
Comment (TsunamiCode5) [inline code-challenges.test.js:138]: I love the refactor AND the documentation, GREAT work Josh. I completely agree with you

File: code-challenges.test.js [hunk 4]
```diff
@@ -20,33 +20,361 @@ const secretCodeWord2 = "Gobbledygook"
...
+// jest-test-3:
+// PASS  ./code-challenges.test.js
+// valueContainsALetter
+//   ✓ return an aray of all the words containing that particular letter (1 ms)
+
+// Test Suites: 1 passed, 1 total
+// Tests:       1 passed, 1 total
 
 // b) Create the function that makes the test pass.
```
Comment (TsunamiCode5) [inline code-challenges.test.js:250]: I would take another look at this one. I copied it into my vscode and it is throwing an error

File: code-challenges.test.js [hunk 5]
```diff
@@ -20,33 +20,361 @@ const secretCodeWord2 = "Gobbledygook"
...
+//     let arr = array.sort() 
+//     if (arr[0] === arr[1] && arr[2] === arr[3] && arr[2] === arr[4] 
+//         || 
+//         arr[0] === arr[1] && arr[0] === arr[2] && arr[3] === arr[4]) {
+//         return true
+//     }  else {
+//         return false
+//     }
+// }
+
```
Comment (TsunamiCode5) [inline code-challenges.test.js:367]: NICEEEE

**Review by TsunamiCode5 — APPROVED** [review]
Josh, absolutely outstanding assessment! You did a great job with your logic in your code and you also did phenomenal applying your pseudocode. The documentation throughout your assessment as well, is solid. You should be very proud of your progress. Your interview answers are so well written and thought through, continue to practice this way because in a stressful environment of a future tech interview you will be able to easily recall techniques like the ones you are implementing here. This assessment comprehension level is competent. Continue the great work and amazing efforts!

File: code-challenges.test.js (hunk 1 above)
Comment (JoshuaMaurer) [reply code-challenges.test.js:29]: Haha! Of course I enjoy you feedback!!! Thank you!!!

File: code-challenges.test.js (hunk 4 above)
Comment (JoshuaMaurer) [reply code-challenges.test.js:250]: Hey Nicole, on this one are you referring to the just the test? I copied it from here right into my vscode and it passed on my end.

**Pull request merged**